src/AnalyticsEventReporter.ts

```diff
@@ -60,22 +60,19 @@ export class AnalyticsEventReporter implements AnalyticsEventService {
       const shouldUseBeacon = useBeacon(finalPayload, this.config.forceFetch);
       const requestUrl = setupRequestUrl(this.config.env, this.config.region);
 
-      // If useBeacon returns true, return boolean response of postWithBeacon
+      // If useBeacon returns true, return boolean response of postWithBeacon as string.
       if (shouldUseBeacon) {
-        return await postWithBeacon(requestUrl, finalPayload);
+        return String(postWithBeacon(requestUrl, finalPayload));
```

Comment:
In this case if postFails, shouldn't we do Promise.reject instead? Can do Promise.reject("Beacon Call Failed")

If it's successful, I think can just do Promise.resolve("");

We should also wrap the postWithBeacon call within a promise, otherwise it won't be async as well. Something like 
```
return new Promise((resolve), (reject) => {
 if(postWithBeacon){
  resolve("");
}else{
  reject("Failed Beacon Call")
}
```